src/pages/settings/Wallet/WalletPage/WalletPage.js

```diff
@@ -0,0 +1,534 @@
+    };
+
+    useEffect(() => {
+        PaymentMethods.openWalletPage();
+    }, []);
+
+    useEffect(() => {
+        // If the user was previously offline, skip debouncing showing the loader
+        if (!network.isOffline) {
+            updateShouldShowLoadingSpinner();
+        } else {
+            debounceSetShouldShowLoadingSpinner();
+        }
+    }, [network.isOffline, debounceSetShouldShowLoadingSpinner, updateShouldShowLoadingSpinner]);
+
+    useEffect(() => {
+        if (network.isOffline) {
+            return;
+        }
+        PaymentMethods.openWalletPage();
+    }, [network.isOffline]);
+
+    useEffect(() => {
+        if (!shouldListenForResize) {
+            return;
+        }
+        setMenuPosition();
+    }, [shouldListenForResize, setMenuPosition]);
+
+    useEffect(() => {
+        if (!shouldShowDefaultDeleteMenu) {
+            return;
+        }
+
+        // We should reset selected payment method state values and close corresponding modals if the selected payment method is deleted
+        let shouldResetPaymentMethodData = false;
+
+        if (paymentMethod.selectedPaymentMethodType === CONST.PAYMENT_METHODS.BANK_ACCOUNT && _.isEmpty(bankAccountList[paymentMethod.methodID])) {
+            shouldResetPaymentMethodData = true;
+        } else if (paymentMethod.selectedPaymentMethodType === CONST.PAYMENT_METHODS.DEBIT_CARD && _.isEmpty(fundList[paymentMethod.methodID])) {
+            shouldResetPaymentMethodData = true;
+        } else if (paymentMethod.selectedPaymentMethodType === CONST.PAYMENT_METHODS.PAYPAL && _.isEmpty(payPalMeData)) {
+            shouldResetPaymentMethodData = true;
+        }
+        if (shouldResetPaymentMethodData) {
+            // Close corresponding selected payment method modals which are open
+            if (shouldShowDefaultDeleteMenu) {
+                hideDefaultDeleteMenu();
+            }
+        }
+    }, [hideDefaultDeleteMenu, paymentMethod.methodID, paymentMethod.selectedPaymentMethodType, bankAccountList, fundList, payPalMeData, shouldShowDefaultDeleteMenu]);
+
+    const isPayPalMeSelected = paymentMethod.formattedSelectedPaymentMethod.type === CONST.PAYMENT_METHODS.PAYPAL;
+    const shouldShowMakeDefaultButton =
+        !paymentMethod.isSelectedPaymentMethodDefault &&
+        Permissions.canUseWallet(betas) &&
+        !isPayPalMeSelected &&
+        !(paymentMethod.formattedSelectedPaymentMethod.type === CONST.PAYMENT_METHODS.BANK_ACCOUNT && paymentMethod.selectedPaymentMethod.type === CONST.BANK_ACCOUNT.TYPE.BUSINESS);
+
+    // Determines whether or not the modal popup is mounted from the bottom of the screen instead of the side mount on Web or Desktop screens
+    const isPopoverBottomMount = anchorPosition.anchorPositionTop === 0 || isSmallScreenWidth;
+
+    return (
+        <ScreenWrapper>
+            {shouldShowEmptyState ? (
+                <WalletEmptyState onAddPaymentMethod={paymentMethodPressed} />
+            ) : (
```

Comment:
I think the other pages that use this pattern are not wrapped in the screenwrapper- perhaps this is causing the top bar color bug?

Reply:
Thanks for the tip @grgia !